backend/apps/advanced_search/views.py

```
@@ -0,0 +1,20 @@
+from .models import SearchEmbedding, UserSearchProfile, SearchAnalytics
+from .serializers import SearchEmbeddingSerializer, UserSearchProfileSerializer, SearchAnalyticsSerializer
+
+class SearchEmbeddingViewSet(viewsets.ModelViewSet):
+    queryset = SearchEmbedding.objects.all()
+    serializer_class = SearchEmbeddingSerializer
+    permission_classes = [IsAuthenticated]
+
+class UserSearchProfileViewSet(viewsets.ModelViewSet):
+    queryset = UserSearchProfile.objects.all()
+    serializer_class = UserSearchProfileSerializer
+    permission_classes = [IsAuthenticated]
```

Ruff (0.15.21)

[warning] Mutable default value for class attribute

(RUF012)